src/libraries/Microsoft.VisualBasic.Core/src/Microsoft.VisualBasic.Core.vbproj

```diff
@@ -9,19 +9,21 @@
     <OptionCompare>Binary</OptionCompare>
     <WarningsNotAsErrors>42025</WarningsNotAsErrors>
     <DefineConstants>$(DefineConstants),LATEBINDING=True</DefineConstants>
-    <DefineConstants Condition="'$(TargetsWindows)' == 'true'">$(DefineConstants),TARGET_WINDOWS=True</DefineConstants>
     <NoWarn>$(NoWarn),CA1052,CA1810,CA2200</NoWarn>
-    <NoWarn Condition="'$(TargetsWindows)' != 'true'">$(NoWarn);CA1823</NoWarn>
     <!-- Avoid unused fields warnings in Unix build -->
     <AssemblyName>Microsoft.VisualBasic.Core</AssemblyName>
     <RemoveIntegerChecks>false</RemoveIntegerChecks>
     <RootNamespace />
     <TargetFrameworks>$(NetCoreAppCurrent);$(NetCoreAppCurrent)-windows</TargetFrameworks>
   </PropertyGroup>
+  <!-- DesignTimeBuild requires all the TargetFramework Derived Properties to not be present in the first property group. -->
   <PropertyGroup>
-    <ILLinkDescriptorsXml Condition="'$(TargetsWindows)' == 'true'">$(MSBuildProjectDirectory)\ILLink\ILLink.Descriptors.Windows.xml</ILLinkDescriptorsXml>
+    <TargetPlatformIdentifier>$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)'))</TargetPlatformIdentifier>
```

Comment:
Instead of duplicating this property everywhere it is used, should we just instead move it to a parent Directory.Build.Props so it is defined already for projects similar to the way TargetsWindows used to?

Reply:
That property relies on `TargetFramework` which isn't available in non cross-targeting builds or during design build. See https://github.com/dotnet/runtime/pull/64500#discussion_r799621575. That means we read from the TargetFramework property in props files.